components/climber.py

```diff
@@ -1,13 +1,45 @@
+import wpilib
+from rev import CANSparkMax
+from ids import SparkMaxIds
+
+
 class ClimberComponent:
     def __init__(self) -> None:
         self.deployed = False
 
     def deploy(self) -> None:
         self.deployed = True
 
+    def execute(self) -> None:
+        if self.deployed:
+
+            def __init__(self) -> None:
+                self.deployed = False
+
+    def deploy(self) -> None:
+        self.deployed = True
+
     def execute(self) -> None:
         if self.deployed:
             # Deploy the climber
-            pass
+            self.climbingmotor = CANSparkMax(
+                SparkMaxIds.climber, CANSparkMax.MotorType.kBrushless
+            )  # using a Neo with 4:1 4:1 3:1 ratio
+            self.limitswitch = wpilib.LimitSwitch(2)
```

Comment:
Reading a limit switch from DIO is probably not the best way of doing this. We can connect a limit switch directly to the spark max. See the [REV docs](https://robotpy.readthedocs.io/projects/rev/en/stable/rev/CANSparkBase.html#rev.CANSparkBase)